Packs/NutanixHypervisor/Integrations/NutanixHypervisor/NutanixHypervisor.py

```diff
@@ -0,0 +1,952 @@
+                storage.capacity_bytes==2;host_nic_ids!=35,host_gpus==x is parsed by Nutanix the following way:
+                Return all hosts s.t (storage.capacity_bytes == 2 AND host_nic_ids != 35) OR host_gpus == x.
+
+    In case response was successful, response will be a of list of hosts details.
+
+    Args:
+        client (Client): Client object to perform request.
+        args (Dict): Demisto arguments.
+
+    Returns:
+        CommandResults.
+    """
+    filter_ = args.get('filter')
+    limit = get_and_validate_int_argument(args, 'limit', minimum=MINIMUM_LIMIT_VALUE, default_value=50)
+    page = get_page_argument(args)
+
+    raw_response = client.get_nutanix_hypervisor_hosts_list(filter_, limit, page)
+
+    if raw_response.get('entities') is None:
+        raise DemistoException('Unexpected response for nutanix-hypervisor-hosts-list command')
+
+    outputs = [{k: v for k, v in copy.deepcopy(raw_output).items() if k not in NUTANIX_HOST_FIELDS_TO_REMOVE} for
+               raw_output in
+               raw_response.get('entities')]
+
+    update_dict_time_in_usecs_to_iso_entries(outputs)
+
+    return CommandResults(
+        outputs_prefix='NutanixHypervisor.Host',
+        outputs_key_field='uuid',
+        outputs=outputs,
+        readable_output=tableToMarkdown('Nutanix Hosts List', create_readable_output(outputs)),
+        raw_response=raw_response
+    )
+
+
+def nutanix_hypervisor_vms_list_command(client: Client, args: Dict):
+    """
+    Gets a list all virtual machines by Nutanix service.
+    Possible filters:
+    - offset: The offset to start retrieving virtual machines.
+    - limit: Maximum number of virtual machines to retrieve.
+    - filter: Retrieve virtual machines that matches the filters given.
+              - Each filter is written in the following way: filter_name==filter_value or filter_name!=filter_value.
+              - Possible combinations of OR (using comma ',') and AND (using semicolon ';'), for Example:
+                machine_type==pc;power_state!=off,ha_priority==0 is parsed by Nutanix the following way:
+                Return all virtual machines s.t (machine type == pc AND power_state != off) OR ha_priority == 0.
+
+    In case response was successful, response will be a list of virtual machines details.
+
+    Args:
+        client (Client): Client object to perform request.
+        args (Dict): Demisto arguments.
+
+    Returns:
+        CommandResults.
+    """
+    filter_ = args.get('filter')
+    offset = get_and_validate_int_argument(args, 'offset', minimum=MINIMUM_OFFSET_VALUE)
+    limit = get_and_validate_int_argument(args, 'limit', minimum=MINIMUM_LENGTH_VALUE, default_value=50)
+
+    raw_response = client.get_nutanix_hypervisor_vms_list(filter_, offset, limit)
+
+    outputs = raw_response.get('entities')
+
+    if outputs is None:
+        raise DemistoException('No entities were found in response for nutanix-hypervisor-vms-list command')
```

Comment:
We are not failing the flow if there are no VMs. If there are no VMs' empty list should be returned and it will be ok, this is just to assure that the response was as expected and entities field is found (even if its empty list its ok)
@ShahafBenYakir